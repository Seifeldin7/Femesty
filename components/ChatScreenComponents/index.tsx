export * from './ChatInputBox'
export * from './ChatMessage'