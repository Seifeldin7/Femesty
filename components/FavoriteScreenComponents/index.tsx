export * from "./FavoriteHzScroller";
export * from "./FavoriteLook";
