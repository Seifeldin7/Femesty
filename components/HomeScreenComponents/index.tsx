export * from "./FilterButton"
export * from "./HeaderTab"
export * from "./HorizontalScroller"
export * from "./LookMedia"
export * from "./SideBar"