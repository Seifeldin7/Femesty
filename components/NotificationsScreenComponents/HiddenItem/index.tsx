import { AntDesign } from "@expo/vector-icons";
import React from "react";
import {
  View,
  Text,
  TouchableHighlight,
} from "react-native";
import { responsiveHeight } from "react-native-responsive-dimensions";

import { styles } from "./styles";
import { componentProps } from "./types";

export const HiddenItem = ({ data, onMarkAsRead }: componentProps) => {
  const height = data.item?.subject_type == "Order" ? responsiveHeight(13) : responsiveHeight(13);
  return (
    <View style={{ ...styles.rowBack, height: height }}>
      <TouchableHighlight
        style={{ ...styles.backRightBtn, height: height }}
        onPress={() => onMarkAsRead(data.item.id)}
      >
        <View style={{ justifyContent: "center", alignItems: "center" }}>
          <AntDesign
            style={styles.check}
            name="checkcircleo"
            size={26}
            color="black"
          />
          <Text style={{ color: "white" }}>Mark as read</Text>
        </View>
      </TouchableHighlight>
    </View>
  );
};
