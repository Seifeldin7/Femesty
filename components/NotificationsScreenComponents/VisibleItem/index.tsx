import React from "react";
import { View, Text } from "react-native";
import { styles } from "./styles";
import { componentProps } from "./types";
import { Avatar } from "../../ThemeComponents/Avatar";
import { responsiveHeight } from "react-native-responsive-dimensions";

export const VisibleItem = ({ data }: componentProps) => {
  const height = data.item.subject_type == "Order" ? responsiveHeight(13) : responsiveHeight(13);
  return (
    <View style={{ ...styles.rowFront, height: height }}>
      <View style={styles.rowFrontVisible}>
        {data.item.is_read ? (
          <View testID="notificationCircle" style={styles.circle}>
          </View>
        ) : null}
        {data.item.subject_type == "Order" ? (
          <View style={{ flexDirection: "row" }}>
            <Avatar imgPath={{ uri: data.item.subject?.product.image1 }} style={{marginRight: 10}} size="medium" />
            <View>
              <Text style={styles.desc}> {data.item.description}</Text>
              <Text style={styles.productName}>
                {" "}
                {data.item.subject?.product.name}
              </Text>
              {data.item.created_at ? (
                <View style={{ flexDirection: "row" }}>
                  <Text style={styles.productName}> Delivered By: </Text>
                  <Text style={styles.date}> {data.item.created_at}</Text>
                </View>
              ) : null}
            </View>
          </View>
        ) : (
          <View style={{ flexDirection: "row" }}>
            <Avatar imgPath={{ uri: data.item.subject?.image1 }} size="medium" />
            <View>
              <Text style={styles.desc}> {data.item.description}</Text>
              <Text style={styles.productName}> {data.item.subject?.name}</Text>
              {data.item.created_at ? (
                <View style={{ flexDirection: "row" }}>
                  <Text style={styles.productName}> Delivered By: </Text>
                  <Text style={styles.date}> {data.item.created_at}</Text>
                </View>
              ) : null}
            </View>
          </View>
        )}
      </View>
    </View>
  );
};
