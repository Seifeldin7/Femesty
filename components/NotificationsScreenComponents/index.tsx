export * from "./HiddenItem";
export * from "./VisibleItem";
