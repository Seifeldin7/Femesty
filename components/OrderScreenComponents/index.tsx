export * from './OrderItem'
export * from './OrderCard'
export * from './OrderList'