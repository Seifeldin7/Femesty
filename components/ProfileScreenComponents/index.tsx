export * from "./ProfileInformation"
export * from "./MenuList"