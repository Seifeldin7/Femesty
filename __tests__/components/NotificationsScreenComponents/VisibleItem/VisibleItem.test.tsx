import React from "react";
import { render } from "react-native-testing-library";
import { ReactTestInstance } from "react-test-renderer";
import { VisibleItem } from "../../../../components/NotificationsScreenComponents";

test('Should render visible item correctly in notifications', () => {
    let data = {
        item: {
            id: 1,
            description: "Item is shipped",
            subject_type: "Order",
            subject: {
                delivery_date: "Monday, 28 September",
                product: {
                    name: 'T-Shirt White'
                }
            },
            is_read: false,
        }
    };
    const tree = render(
        <VisibleItem data={data} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

describe('<Notification rendered data (Order)>', () => {
    let queryByText: (text: string) => ReactTestInstance | null;
    let queryByTestId: (text: string) => ReactTestInstance | null;
    let data = {
        item: {
            id: 1,
            description: "Item is shipped",
            subject_type: "Order",
            subject: {
                delivery_date: "Monday, 28 September",
                product: {
                    name: 'T-Shirt White'
                }
            },
            is_read: true,
        }
    };
    beforeEach(() => {
        ({ queryByText, queryByTestId } = render(
            <VisibleItem data={data} />
        ));
    });

    test('Notification description', () => {
        const description = queryByText(data.item.description);
        expect(description).toBeDefined();
    });

    test('Notification product name', () => {
        const name = queryByText(data.item.subject.product.name);
        expect(name).toBeDefined();
    });

    test('Notification date', () => {
        const date = queryByText(data.item.subject.delivery_date);
        expect(date).toBeDefined();
    });

    test('Notification is_read circle', () => {
        const is_read = queryByTestId("notificationCircle");
        expect(is_read).toBeDefined();
    });
});

describe('<Notification rendered data (Product)>', () => {
    let queryByText: (text: string) => ReactTestInstance | null;
    let queryByTestId: (text: string) => ReactTestInstance | null;
    let data = {
        item: {
            id: 1,
            description: "Item is shipped",
            subject_type: "Product",
            subject: {
                name: 'T-Shirt White'
            },
            is_read: false,
        }
    };
    beforeEach(() => {
        ({ queryByText, queryByTestId } = render(
            <VisibleItem data={data} />
        ));
    });

    test('Notification description', () => {
        const description = queryByText(data.item.description);
        expect(description).toBeDefined();
    });

    test('Notification is_read circle', () => {
        const is_read = queryByTestId("notificationCircle");
        expect(is_read).toBeNull();
    });
});
