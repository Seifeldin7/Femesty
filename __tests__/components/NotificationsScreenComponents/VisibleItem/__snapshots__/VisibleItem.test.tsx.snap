// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render visible item correctly in notifications 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 5,
      "elevation": 10,
      "height": 110,
      "margin": 5,
      "marginBottom": 15,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.9,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 10,
        "height": 60,
        "marginBottom": 15,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "marginRight": 5,
            "width": 50,
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "color": "#1E1E1E",
              "fontFamily": "Lato_SemiBold",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
           
          Item is shipped
        </Text>
        <Text
          style={
            Object {
              "color": "#1E1E1E",
              "fontFamily": "Lato_SemiBold",
              "fontSize": 15,
              "marginBottom": 5,
            }
          }
        >
           
          T-Shirt White
        </Text>
      </View>
    </View>
  </View>
</View>
`;
